qiime2/metadata/metadata.py

```diff
@@ -1036,7 +952,42 @@ def name(self):
         """
         return self._series.name
 
-    def __init__(self, series):
+    @property
+    def artifacts(self):
+        """Artifact that is the source of the column.
+
+        This property is read-only.
+
+        Returns
+        -------
+        tuple of qiime2.Artifact
+            Source artifact of the column.
+
+        """
+        if self._source_metadata is None:
+            return ()
+        if self._source_metadata._source_artifact is not None:
+            return (self._source_metadata._source_artifact,)
+
+        old_md = self._source_metadata
+        new_md = self._source_metadata._column_sources[self._series.name]
+        for new_name, old_name in old_md._column_names.items():
+            if new_md is None:
+                if old_md._source_artifact is not None:
+                    return (old_md._source_artifact,)
+                return ()
+            if new_name in new_md._column_sources:
+                old_md = new_md
+                new_md = new_md._column_sources[new_name]
+            elif old_name in new_md._column_sources:
+                old_md = new_md
+                new_md = new_md._column_sources[old_name]
+            else:
```

Comment:
Like the comment says, if we somehow reach this point in code something went very wrong, I'm thinking we just let that crash and burn (it shouldn't even be possible anyway), but something made me think putting this here might be worthwhile.

Reply:
I haven't looked closely at this code, but if this endpoint represents a terminal failure condition, I think it would be best to raise an exception.